analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/PermissiveCors.cs

```diff
@@ -0,0 +1,148 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Extensions;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.Helpers.Trackers;
+
+namespace SonarAnalyzer.Rules.Hotspots
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class PermissiveCors : TrackerHotspotDiagnosticAnalyzer<SyntaxKind>
+    {
+        private const string DiagnosticId = "S5122";
+        private const string MessageFormat = "Make sure this permissive CORS policy is safe here.";
+        private const string AccessControlAllowOriginHeader = "Access-Control-Allow-Origin";
+        private const string AccessControlAllowOriginPropertyName = "AccessControlAllowOrigin";
+        private const string StarConstant = "*";
+        private const int ParameterCount = 2;
```

Comment:
this is just for `Add`/`Append`, not for `WithOrigins` or `AllowAnyOrigin` - so it's not "the parameter count for all methods verified in this rule"... 

and this could be a `const int` just inside `SetupInvocationTracker`

Reply:
Yes, [WithOrigins](https://docs.microsoft.com/en-us/dotnet/api/Microsoft.AspNetCore.Cors.Infrastructure.CorsPolicyBuilder.WithOrigins?view=aspnetcore-5.0&viewFallbackFrom=netcore-5.0) needs `params string[] origins` [AllowAnyOrigin](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.cors.infrastructure.corspolicybuilder.allowanyorigin?view=aspnetcore-5.0) does not has any parameters